File: nineteen.rb
require "pry"
twentieth_century = []
week = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"]

months = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]

1.upto(100) do |year|
  months.each do |month|
    if month === 28 && (year > 3 && year % 4 === 0)
      1.upto(29) { |day| twentieth_century << day }
    else
      1.upto(month) { |day| twentieth_century << day }
    end
  end
end

sundays = 0

twentieth_century.each_with_index do |day, ind|
  sunday = ind % 7 === 0
  first = day === 1
  sundays += 1 if first && sunday
end

puts sundays

File: nineteen.js
var months = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
var twentiethCentury = []

for (var year = 1; year < 100; year++) {
  months.forEach(function(month) {
    var feb = month === 28
    var leapYear = year > 3 && year % 4 === 0
    if (feb & leapYear) {
      for (var day = 1; day < 30; day++) {
        twentiethCentury.push(day)
      }
    }
    else {
      for (var day = 1; day <= month; day++) {
        twentiethCentury.push(day)
      }
    }
  })
}

var firstSundays = 0
var i = 1

twentiethCentury.forEach(function(day) {
  var first = day === 1
  var sunday = i % 7 === 0
  if (first && sunday) {
    firstSundays++
  }
  i++
})

console.log(firstSundays)
